pandas/core/arrays/sparse/array.py

```diff
@@ -1179,6 +1179,13 @@ def copy(self: SparseArrayT) -> SparseArrayT:
         values = self.sp_values.copy()
         return self._simple_new(values, self.sp_index, self.dtype)
 
+    def _values_for_argsort(self) -> np.ndarray:
+        return self._sparse_values
+
+    def _mask_for_argsort(self) -> np.ndarray:
+        mask = ExtensionArray._mask_for_argsort(self)
+        return mask[self._sparse_index.indices]
```

Comment:
See other comments - why not do this directly instead of dispatching `SparseArray -> ExtensionArray -> SparseArray`